packages/devextreme/testing/tests/DevExpress.ui.widgets.editors/datebox.tests.js

```diff
@@ -5760,6 +5760,48 @@ QUnit.module('datebox validation', {}, () => {
     });
 });
 
+[
+    { type: 'time', input: 'p', displayFormat: 'HH:mm a', expectedValue: 'PM' },
+    { type: 'time', input: 'a', displayFormat: 'HH:mm a', expectedValue: 'AM' },
+    { type: 'time', input: 'x', displayFormat: 'HH:mm a', expectedValue: null },
+    { type: 'time', input: 'y', displayFormat: 'HH:mm a', expectedValue: null },
+    { type: 'time', input: 'z', displayFormat: 'HH:mm a', expectedValue: null },
+    { type: 'datetime', input: 'p', displayFormat: 'dd/MM/yyyy HH:mm a', expectedValue: 'PM' },
+    { type: 'datetime', input: 'a', displayFormat: 'dd/MM/yyyy HH:mm a', expectedValue: 'AM' },
+    { type: 'datetime', input: 'x', displayFormat: 'dd/MM/yyyy HH:mm a', expectedValue: null },
+    { type: 'datetime', input: 'y', displayFormat: 'dd/MM/yyyy HH:mm a', expectedValue: null },
+    { type: 'datetime', input: 'z', displayFormat: 'dd/MM/yyyy HH:mm a', expectedValue: null },
+].forEach(({ type, input, displayFormat, expectedValue }) => {
+    QUnit.module(`DateBox AM/PM Handling - ${type} datebox type - ${displayFormat} a display format`, {
+        beforeEach: function() {
+            this.$element = $('#dateBox').dxDateBox({
+                value: new Date('10/10/2012 13:07'),
+                useMaskBehavior: true,
+                type: type,
+                mode: 'text',
+                displayFormat: displayFormat,
+                pickerType: 'calendar'
```

Comment:
please check if all these options are required for testing current scenario. eg pickerType